cli/azd/extensions/azure.ai.agents/internal/project/service_target_agent.go

```diff
@@ -516,10 +516,23 @@ func (p *AgentServiceTargetProvider) deployHostedAgent(
 	}
 
 	// Step 3: Create agent request with image URL and resolved environment variables
+	var foundryAgentConfig *ServiceTargetAgentConfig
+	if err := UnmarshalStruct(serviceConfig.Config, &foundryAgentConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse foundry agent config: %w", err)
+	}
+
+	var cpu, memory string
+	if foundryAgentConfig.Container != nil && foundryAgentConfig.Container.Resources != nil {
+		cpu = foundryAgentConfig.Container.Resources.Cpu
+		memory = foundryAgentConfig.Container.Resources.Memory
+	}
+
 	request, err := agent_yaml.CreateAgentAPIRequestFromDefinition(
 		agentDef,
 		agent_yaml.WithImageURL(fullImageURL),
 		agent_yaml.WithEnvironmentVariables(resolvedEnvVars),
+		agent_yaml.WithCPU(cpu),
+		agent_yaml.WithMemory(memory),
```

Comment:
When `cpu` and `memory` are not defined in `azure.yaml` you would see empty strings in here.
Make sure that empty strings are supported by the request you send to the service.

It is different to do-not  include a field `cpu` than sending a cpu with an empty string - as a way to declare that the user did not set any override for it.